plugins/inputs/sql/README.md

```diff
@@ -0,0 +1,192 @@
+			sanitize = false				# true: will perform some chars substitutions (false: use value as is)
+			ignore_row_errors				# true: if an error in row parse is raised then the row will be skipped and the parse continue on next row (false: fatal error)
+```
+sql_script is read only once, if you change the script you need to restart telegraf
+
+## Field names
+Field names are the same of the relative column name or taken from value of a column. If there is the need of rename the fields, just do it in the sql, try to use an ' AS ' .
+
+## Datatypes:
+Using field_cols list the values are converted by the go database driver implementation. 
+In some cases this automatic conversion is not what we expect, therefore you can force the destination datatypes specifing the columns in the bool/int/float/time_fields lists, then if possible the plugin converts the data.
+All field lists can contain an regex for column name matching.
+If an error in conversion occurs then telegraf exits, therefore a --test run is suggested.
+
+## Tested Databases
+Actually I run the plugin using oci8,mysql,mssql,postgres,sqlite3
+
+
+## Example for collect multiple counters defined as COLUMNS in a table (vertical counter structure):
+Here we read a table where each counter is on a different row. Each row contains a column with the name of the counter (counter_name) and a column with his value (cntr_value) and some other columns that we use as tags  (instance_name,object_name)
+
+###Config
+```
+[[inputs.sql]]
+	interval = "60s"
+	driver = "mssql"
+	servers = [
+		"Server=mssqlserver1.my.lan;Port=1433;User Id=telegraf;Password=secret;app name=telegraf"
+		"Server=mssqlserver2.my.lan;Port=1433;User Id=telegraf;Password=secret;app name=telegraf"
+	]
+	hosts=["mssqlserver_cluster_1","mssqlserver_cluster_2"]
+
+	[[inputs.sql.query]]
+		measurement = "os_performance_counters"
+		ignore_other_fields=true
+		sanitize=true
+		query="SELECT * FROM sys.dm_os_performance_counters WHERE object_name NOT LIKE '%Deprecated%' ORDER BY counter_name"
+		tag_cols=["instance_name","object_name"]
+		field_name = "counter_name"
+		field_value = "cntr_value"
+```
+### Result:
+```
+> os_performance_counters,host=mssqlserver_cluster_1,object_name=MSSQL$TESTSQL2014:Broker_Statistics Activation_Errors_Total=0i 1494496261000000000
+> os_performance_counters,host=mssqlserver_cluster_1,object_name=MSSQL$TESTSQL2014:Cursor_Manager_by_Type,instance_name=TSQL_Local_Cursor Active_cursors=0i 1494496261000000000
+> os_performance_counters,instance_name=TSQL_Global_Cursor,host=mssqlserver_cluster_1,object_name=MSSQL$TESTSQL2014:Cursor_Manager_by_Type Active_cursors=0i 1494496261000000000
+> os_performance_counters,host=mssqlserver_cluster_1,object_name=MSSQL$TESTSQL2014:Cursor_Manager_by_Type,instance_name=API_Cursor Active_cursors=0i 1494496261000000000
+> os_performance_counters,host=mssqlserver_cluster_1,object_name=MSSQL$TESTSQL2014:Cursor_Manager_by_Type,instance_name=_Total Active_cursors=0i 1494496261000000000
```

Comment:
If we kept vertical structure, we would need to combine these into a single "metric" so that it would behave properly with existing buffer sizes as well as streaming to Kapacitor.